src/devices/bus/hp80_io/82937.h

```diff
@@ -0,0 +1,71 @@
+// license:BSD-3-Clause
+// copyright-holders: F. Ulivi
+/*********************************************************************
+
+    82937.h
+
+    82937 module (HPIB interface)
+
+*********************************************************************/
+
+#ifndef MAME_BUS_HP80_IO_82937_H
+#define MAME_BUS_HP80_IO_82937_H
+
+#pragma once
+
+#include "hp80_io.h"
+#include "cpu/mcs48/mcs48.h"
+#include "bus/ieee488/ieee488.h"
+#include "machine/1mb5.h"
+
+class hp82937_io_card_device : public hp80_io_card_device
+{
+public:
+	// construction/destruction
+	hp82937_io_card_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	virtual ~hp82937_io_card_device();
+
+	virtual void install_read_write_handlers(address_space& space , uint16_t base_addr) override;
+
+	virtual void inten() override;
+	virtual void clear_service() override;
+
+	DECLARE_WRITE_LINE_MEMBER(reset_w);
+	DECLARE_READ_LINE_MEMBER(t0_r);
+	DECLARE_READ8_MEMBER(p1_r);
+	DECLARE_WRITE8_MEMBER(p1_w);
+	DECLARE_READ8_MEMBER(dio_r);
+	DECLARE_WRITE8_MEMBER(dio_w);
+	DECLARE_READ8_MEMBER(switch_r);
+	DECLARE_WRITE8_MEMBER(latch_w);
+	DECLARE_WRITE_LINE_MEMBER(ieee488_ctrl_w);
```

Comment:
Read/write members used from your own `MACHINE_CONFIG_MEMBER` functions shouldn't really be public if the signals aren't exposed outside the device itself (stuff used in address maps still needs to be public because of how address maps are implemented).